packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -5,9 +5,57 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React from 'react'
+import React, {useEffect, useRef} from 'react'
+import PropTypes from 'prop-types'
+import {useLocation} from 'react-router-dom'
+import {uuidv4} from '../../../utils/uuidv4.client'
+
 import DeviceContext from '../device-context'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+const CorrelationIdContext = React.createContext()
+// regexp for uuid format
+const regexp = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i
```

Comment:
So I left a comment about this format checking before. I'm assuming that you are doing this because the CommerceSDKReact provider only works with v4 uuids? 

If that is the case, there isn't really anything stopping the end used changing what gets passed to the provider since it's just a prop. So I think it's somewhat the responsibility of the commerce-sdk-isomorphic to ensure they only accept v4 id's. 

Also, adding this type checking is going to make writing tests just that more hard.

But if you have a strong argument for having this here, please let me know. At a minimum we can probably do this check outside of the useEffect similar to the below comment.

Reply:
Got it, I don't have a very strong argument about leaving it in. I'll make the change.